src/extension.ts

```diff
@@ -221,4 +221,130 @@ export function activate(
+    if ((sourcegraph as any)?.app?.registerFileDecorationProvider) {
+        interface FileDecoration {
+            uri: string
+            after?: {
+                contentText: string
+                color: string
+                hoverMessage?: string
+            }
+            meter?: {
+                value: number
+                min?: number
+                max?: number
+                low?: number
+                high?: number
+                optimum?: number
+                hoverMessage?: string
+            }
+        }
+
+        context.subscriptions.add(
+            (sourcegraph as any).app.registerFileDecorationProvider({
+                provideFileDecorations: async ({
+                    files,
+                    uri,
+                }: {
+                    uri: string
+                    files: {
+                        uri: string
+                        isDirectory: boolean
+                        name: string
+                        path: string
+                    }[]
+                }) => {
+                    const { repo, rev } = resolveDocumentURI(uri)
+                    const apiParams = codecovParamsForRepositoryCommit({ repo, rev }, sourcegraph)
+
+                    // Fetch commit coverage to get ratio for files, tree coverage to get ratio for directories
+                    const nonDirectoryFiles = files.filter(file => !file.isDirectory)
+                    const directories = files.filter(file => file.isDirectory)
+                    const [commitCoverage, ...treeCoverages] = await Promise.allSettled([
+                        getCommitCoverage(apiParams),
+                        ...directories.map(async (dir, i) => {
+                            const treeCoverage = await getTreeCoverage({ ...apiParams, path: dir.path })
+                            return { treeCoverage, directory: directories[i] }
+                        }),
+                    ])
+
+                    // Iterate over files and get value from commit coverage to construct file decorations
+                    const fileDecorations: FileDecoration[] = []
+
+                    if (commitCoverage.status === 'fulfilled' && commitCoverage.value) {
+                        const { files: reportFiles } = commitCoverage.value.commit.report
+                        for (const file of nonDirectoryFiles) {
+                            const report = reportFiles[file.path]
+                            if (!report) {
+                                continue
+                            }
+
+                            const ratio = parseInt(report.t.c, 10)
+
+                            fileDecorations.push({
+                                uri: file.uri,
+                                after: {
+                                    contentText: `${ratio}%`,
+                                    color: 'gray',
+                                    hoverMessage: `the coverage ratio is ${ratio}%`,
```

Comment:
This tooltip doesn't match the design: https://www.figma.com/file/Bb0SA1zYNVQSXakzkMVhpH/Codecov-file-decorations-721?node-id=415%3A2592
Was that deviation intentional?

And is the "Codecov:" prefix added by the web app? That may be a good idea to make sure meters are always tracable back to a specific extension

Reply:
Yeah, "Codecov: code coverage" seems somewhat redundant. 

I think prefixing in the web app is a great idea. How about "coverage ratio $RATIO", then the end result will be "Codecov: coverage ratio $RATIO"?

Reply:
Sounds good

Reply:
I would suggest "Codecov: xx% covered" ("ratio" sounds a bit too technical imo)